apple/header_map.bzl

```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Rules for creating header maps.
+"""
+
+load(
+    "@build_bazel_rules_apple//apple/internal:header_map.bzl",
+    _header_map = "header_map",
+)
+
+header_map = _header_map
```

Comment:
I was thinking we'd want this to be public so it could be used by other external macros, etc. We can definitely automatically support it with `experiemental_mixed_language_library` but it might still be wanted by folks using a pure `objc_library` for example.

Reply:
As someone who has recently adapted (copied) this work into their own project to create a functioning mixed-language library, I would very much appreciate it being made part of the public API. Though, that was under the current constraint that headermaps don't propagate correctly without manual configuration, which I recall there has been discussion towards addressing.

Reply:
👍